WalletWasabi/Services/UpdateChecker.cs

```diff
@@ -15,9 +14,9 @@ public class UpdateChecker : PeriodicRunner
 
 		public UpdateChecker(TimeSpan period, WasabiSynchronizer synchronizer) : base(period)
 		{
-			Synchronizer = Guard.NotNull(nameof(synchronizer), synchronizer);
+			Synchronizer = synchronizer;
 			WasabiClient = synchronizer.WasabiClient;
-			UpdateStatus = new UpdateStatus(true, true, new Version(), 0);
+			_updateStatus = new UpdateStatus(true, true, new Version(), 0);
```

Comment:
This is not how we do it usually.

Reply:
True. But I would argue that this is actually the correct way as it removes a nullability warning. I don't think there is any other reasonable way to remove that nullability warning. 

The original version assigns `UpdateStatus` property which does not fire any event so the effect is the same.

Reply:
To explain myself better: If there is a backing field for a property, I think it's correct to initialize that backing field and not the property in constructor.

Reply:
IMO we should name private properties without a prefix and initialize them like this: `this.updateStatus = updateStatus`. It seems to be the .NET standard.

But we aren't doing that, so I don't do that either. I'm fine with having a discussion on backing field initialization, but it's better to be consistent with the codebase unless you plan to change everything.

Reply:
> IMO we should name private properties without a prefix and initialize them like this: `this.updateStatus = updateStatus`. It seems to be the .NET standard.

I like that more in C# too. `_fieldName` looks more like C++ to me.



> But we aren't doing that, so I don't do that either. I'm fine with having a discussion on backing field initialization, but it's better to be consistent with the codebase unless you plan to change everything.

I think it's a worthy goal to initialize backing fields for properties in constructors. It would remove a lot of nullability warnings. Staying with status quo means that you can't get rid of those warnings AFAIK. 

I'm not sure whether it's easy to identify all those "bad" initializations easily but I guess that finding with regexp can help  pinpoint those occurrences. Still, MSVS warns about these things so I think we can do it gradually too (and not risk too much).

Reply:
> IMO we should name private properties without a prefix and initialize them like this: this.updateStatus = updateStatus. It seems to be the .NET standard.

Most of dotnet repos say internal or private fields should be `_camelCase`:
https://github.com/dotnet/runtime/blob/839b1343fbad61526fedcaa722f1ae48b298c543/.editorconfig#L75-L82
https://github.com/dotnet/roslyn/blob/81b6a510309c8fec3c299b281d85d820c108eef5/.editorconfig#L107-L115
https://github.com/dotnet/efcore/blob/4ddfde1cecca9c88bdee48120ef29e0039372138/.editorconfig#L235-L241
https://github.com/dotnet/wpf/blob/fc640a562c3090fe8cc7541dfb3550181bbf4f91/.editorconfig#L73-L82
https://github.com/dotnet/winforms/blob/da4cd35695fc7fbd19adfc3067f7b44ac90bc9e0/.editorconfig#L52-L61

Reply:
> I'm not sure whether it's easy to identify all those "bad" initializations easily but I guess that finding with regexp can help pinpoint those occurrences.

#4578 removes some of those warnings.

Reply:
> #4578 removes some of those warnings.

Good job 👍

Reply:
How about this? https://github.com/zkSNACKs/WalletWasabi/pull/4590

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/4590#issuecomment-712147907 - my comment